core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/jdbc/JDBCUserStoreManager.java

```diff
@@ -2568,10 +2576,9 @@ private void updateStringValuesToDatabase(Connection dbConnection, String sqlStm
                 dbConnection.commit();
             }
         } catch (SQLException e) {
+            DatabaseUtil.rollBack(dbConnection);
             String msg = "Error occurred while updating string values to database.";
-            if (log.isDebugEnabled()) {
-                log.debug(msg, e);
-            }
+            log.error(msg, e);
```

Comment:
In here also, put a comment including the reason.

Reply:
Fixed https://github.com/wso2/carbon-kernel/pull/2810/commits/66bfa2016cccc221cb3ad7fbcd1a354bd6ab6cb2